### DISCUSSION TITLE
Rofi Alt+Tab working flawlessly

### DISCUSSION THREAD

**Discussion by wellzenon** [davatorium/rofi#1862] [Show and tell]
Hey, I don't know if this is really the place, but I found a solution for the [issue 38](https://github.com/davatorium/rofi/issues/38) which is now closed.

```bash
          rofi -modi window -show window -hide-scrollbar -padding 50 -line-padding 4 -auto-select \
          -kb-cancel "Alt+Escape,Escape" \
          -kb-accept-entry "!Alt-Tab,!Alt+Alt_L,Return"\
          -kb-row-down "Alt+Tab,Alt+Down" \
          -kb-row-up "Alt+ISO_Left_Tab,Alt+Up" \
          -selected-row 1 &
          while for did in $(xinput --list --id-only) ; do xinput query-state $did 2>/dev/null | grep down ; done | egrep -q . ; do sleep 0 ; done
          xdotool key --delay 0 Enter
```
The solution is a mix from https://github.com/davatorium/rofi/issues/38#issuecomment-596037267 and https://superuser.com/a/1602025

**Comment by DaveDavenport** [comment]
you can do a no-action timeout to accept entry now in rofi directly ( see 1.7.0 release notes and/or documentation)

```css
configuration {
  timeout {
      delay:  5;
      action: "kb-accept-entry";
  }
}
```

**Reply by wellzenon** [reply]
That's much better!

**Reply by DaveDavenport** [reply]
```bash
rofi -timeout-delay 5 -timeout-action "kb-accept-entry"
```
if you want to pass on cmdline.

**Reply by wellzenon** [reply]
`rofi -timeout-delay 1 -timeout-action "kb-accept-entry"`
did it for me, thank you! I couldn't get it working through the config file, though, and the style got reseted. Looking into it to find out what happened

**Reply by bartimois** [reply]
Hey, would it be possible to kind of Sort the list as in recently used? So that it kind of behaves like the Windows/kde alt+Tab

cause right now e. g. Im in my browser which is in the second row of rofi, and my terminal in the first. Now  when i alt+tab, it moves to opera again, so i basically have to tab twice or even more (depending on current clients). I could use a shortcut to go up one row but this doesnt seem like a clean solution to me.

EDIT: also I use DWM cause i thought the WM could set the sort but i couldnt find anything at all.

**Reply by DaveDavenport** [reply]
1.) why are you adding a different question to this thread?
2.) See relevant issues on this (https://github.com/davatorium/rofi/issues/641)
